### PR TITLE
JSON input format and InputProcessor refactor

### PR DESCRIPTION
**Pull request by mbadams5**
This introduces JSON as the new preferred input format. EnergyPlus will still accept IDF input for a TBD number of future releases. Developers will add and change the Energy+.idd until IDF is no longer accepted for input. There is a python script that is built into the build process that converts the IDD to JDD automatically. 

To reduce the amount of code changes needed, all InputProcessor functions maintain their signatures and return the same data format, however, they parse and query against the JSON data structure internally. Through our refactoring efforts, we found two functions in InputProcessor that causes O( n^2 ) complexity and were able to change it to O( n ), due to amortized O(1) lookup with `unordered_map`. The table below shows this speed up for two different files; one reference building and one worst case input with an extreme number of surfaces (45,382 in 80 zones vs 871 in 118 zones). The table also shows the speed up in ProcessInput, parsing input, and parsing schema.

|  | Outpatient |  |  | prj10 |  |  |
| :-: | :-: | :-: | :-: | :-: | :-: | :-: |
|  | E+ 8.5 Release | PR - IDF | PR - JSON | E+ 8.5 Release | PR - IDF | PR - JSON |
| ProcessInput | 366 | 300 (18%) | 234 (36%) | 4322 | 3355 (22%) | 1637 (62%) |
| GetSurfaceData | 28 | 13 (54%) | 13 (54%) | 72688 | 21001 (71%) | 21001 (71%) |
| GetObjectItem | 38 | 29 (24%) | 29 (24%) | 41617 | 333 (99.2%) | 333 (99.2%) |
| VerifyName | 2 | 0 (100%) | 0 (100%) | 11055 | 5 (99.9%) | 5 (99.9%) |
| Parse IDF | 174 | 135 (22%) | 69 (60%) | 4130 | 3190 (23%) | 1472 (64%) |
| Parse IDD | 192 | 165 (14%) | 165 (14%) | 192 | 165 (14%) | 165 (14%) |

We are using the [JSON for Modern C++](https://github.com/nlohmann/json) library to parse and our own custom JSON Schema validator for validation. 
### **\* NOTE ***

This requires the update to MSVC 2015 so we need to move CI machines and windows developers to Visual Studio 2015. This should be done anyways since MSVC 2015 is [C++11 feature complete](https://msdn.microsoft.com/en-us/library/hh567368.aspx).
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] Open windows IDF Editor with modified IDD to check for errors
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] Changed docs build successfully
- [ ] ExpandObjects changes?
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -40652,6 +40665,9 @@ ZoneHVAC:AirDistributionUnit,
        \key AirTerminal:DualDuct:ConstantVolume
        \key AirTerminal:DualDuct:VAV
        \key AirTerminal:SingleDuct:ConstantVolume:Reheat
+       \key AirTerminal:SingleDuct:ConstantVolume:FourPipeBeam
+       \key AirTerminal:SingleDuct:InletSideMixer
+       \key AirTerminal:SingleDuct:SupplySideMixer
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: AirTerminal:SingleDuct:InletSideMixer and ...:SupplySideMixer were combined into AirTerminal:SingleDuct:Mixer.

File: idd/Energy+.idd.in [hunk 2]
```diff
@@ -73151,7 +73193,6 @@ SetpointManager:OutdoorAirPretreat,
        \memo produce the reference setpoint condition at the
        \memo mixed air node when mixed with the return air stream
   A1 , \field Name
-       \required-field
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: This seems like a mistake to delete \required-field?

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -87021,7 +87064,7 @@ UtilityCost:Qualify,
         \memo "Qualified" entry will say "No" and the UtilityCost:Qualify that caused its exclusion
         \memo is shown. Multiple UtilityCost:Qualify objects can appear for the same tariff and
         \memo they can be based on any variable.
-   A1,  \field Name
+   A1,  \field Utility Cost Qualify Name
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: Just curious why some Name fields have been renamed to more specific field names.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 4]
```diff
@@ -2202,47 +2200,18 @@ namespace FanCoilUnits {
 
 			if ( CurDeadBandOrSetback( ZoneNum ) || AirMassFlow < SmallMassFlow ) UnitOn = false;
 
-			// zero the hot & cold water flows
-			mdot = 0.0;
-			SetComponentFlowRate( mdot, FanCoil( FanCoilNum ).ColdControlNode, FanCoil( FanCoilNum ).ColdPlantOutletNode, FanCoil( FanCoilNum ).CWLoopNum, FanCoil( FanCoilNum ).CWLoopSide, FanCoil( FanCoilNum ).CWBranchNum, FanCoil( FanCoilNum ).CWCompNum );
```
Comment (mjwitte) [inline src/EnergyPlus/FanCoilUnits.cc]: Lots of extra diffs showing in fancoil - did something get lost in a merge?  Need to double-check this file.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 5]
```diff
@@ -2263,71 +2243,51 @@ namespace FanCoilUnits {
...
-					if ( FirstHVACIteration ) Par( 2 ) = 1.0;
-					Par( 3 ) = ControlledZoneNum;
-					Par( 4 ) = QZnReq;
-					Par( 5 ) = double( FanCoil( FanCoilNum ).ColdControlNode );
-					SolveRegulaFalsi( 0.001, MaxIterCycl, SolFlag, PLR, CalcFanCoilPLRResidual, 0.0, 1.0, Par );
-					if ( SolFlag == -1 ) {
-						// tighten limits on water flow rate to see if this allows convergence
-						CoolingLoad = true;
-						HeatingLoad = false;
-						TightenAirAndWaterFlowLimits( FanCoilNum, CoolingLoad, HeatingLoad, FanCoil( FanCoilNum ).ColdControlNode, ControlledZoneNum, FirstHVACIteration, QZnReq, PLRMin, PLRMax );
```
Comment (mjwitte) [inline src/EnergyPlus/FanCoilUnits.cc]: Looks like these changes are getting squashed.

**Review by mjwitte — COMMENTED** [review]
Just some initial comments from browsing the file diffs on github  If I'm following correctly, the only changes in source files outside the input processor are to eliminate "Using InputProcessor:*" and some whitespace cleanup.  FanCoilUnit.cc appears to have additional unwanted changes.  Other source files not displayed here will need an offline review to check for any other extra changes.

**Comment by JasonGlazer** [comment]
Lots of diffs, maybe something in the defaulting?

File: idd/Energy+.idd.in (hunk 1 above)
Comment (mbadams5) [reply idd/Energy+.idd.in]: Probably merging error. Will address.

File: idd/Energy+.idd.in (hunk 3 above)
Comment (mbadams5) [reply idd/Energy+.idd.in]: So objects where the Name fields were renamed was due to how our automated translation from IDF to JDF works. If there is a Name field, it becomes a key and thus must be unique by the JSON standard. What happens is that in some cases, objects have a Name field but the name can be non-unique. Thus the easiest and most straightforward way to address this was to change the Name field to a more specific field name.

File: src/EnergyPlus/FanCoilUnits.cc (hunk 4 above)
Comment (mbadams5) [reply src/EnergyPlus/FanCoilUnits.cc]: Yes there might have been a merging error. Will address.

File: src/EnergyPlus/FanCoilUnits.cc (hunk 5 above)
Comment (mbadams5) [reply src/EnergyPlus/FanCoilUnits.cc]: I will look into this.

**Comment by JasonGlazer** [comment]
@mbadams5 Could you include some JDF's in your next commit as examples?

**Comment by JasonGlazer** [comment]
@mbadams5 The JDD (replacement of the IDD file) seems to have a lot of redundancy in it. Just looking at the ElectricEquipment object (currently line 72717), the names of fields seem to be repeated two or three times. Is this the final format that is going to be used for the JDD?

```
        "ElectricEquipment": {
            "name": {
                "is_required": true, 
                "type": "string", 
                "reference": [
                    "ElectricEquipmentNames"
                ]
            }, 
            "min_fields": 10.0, 
            "patternProperties": {
                ".*": {
                    "required": [
                        "zone_or_zonelist_name", 
                        "schedule_name"
                    ], 
                    "properties": {
                        "zone_or_zonelist_name": {
                            "type": "string", 
                            "object_list": [
                                "ZoneAndZoneListNames"
                            ], 
                            "data_type": "object_list"
                        }, 
                        "watts_per_zone_floor_area": {
                            "units": "W/m2", 
                            "ip-units": "W/ft2", 
                            "minimum": 0.0, 
                            "type": "number"
                        }, 
                        "watts_per_person": {
                            "units": "W/person", 
                            "ip-units": "W/person", 
                            "minimum": 0.0, 
                            "type": "number"
                        }, 
                        "schedule_name": {
                            "note": "units in schedule should be fraction applied to design level of electric equipment, generally (0.0 - 1.0)", 
                            "type": "string", 
                            "object_list": [
                                "ScheduleNames"
                            ], 
                            "data_type": "object_list"
                        }, 
                        "design_level": {
                            "units": "W", 
                            "ip-units": "W", 
                            "minimum": 0.0, 
                            "type": "number"
                        }, 
                        "design_level_calculation_method": {
                            "note": "The entered calculation method is used to create the maximum amount of electric equipment for this set of attributes Choices: EquipmentLevel => Equipment Level -- simply enter watts of equipment Watts/Area => Watts per Zone Floor Area -- enter the number to apply.  Value * Floor Area = Equipment Level Watts/Person => Watts per Person -- enter the number to apply.  Value * Occupants = Equipment Level", 
                            "default": "EquipmentLevel", 
                            "enum": [
                                "EquipmentLevel", 
                                "Watts/Area", 
                                "Watts/Person"
                            ], 
                            "type": "string"
                        }, 
                        "fraction_lost": {
                            "default": 0.0, 
                            "minimum": 0.0, 
                            "type": "number", 
                            "maximum": 1.0
                        }, 
                        "end_use_subcategory": {
                            "note": "Any text may be used here to categorize the end-uses in the ABUPS End Uses by Subcategory table. The following special tags will also place the end-use in specific rows in the LEED Summary table EAp2-4/5. Performance Rating Method Compliance: Fans-Parking Garage, Interior Lighting-Process, Cooking, Industrial Process, Elevators and Escalators.", 
                            "retaincase": true, 
                            "default": "General", 
                            "type": "string"
                        }, 
                        "fraction_latent": {
                            "default": 0.0, 
                            "minimum": 0.0, 
                            "type": "number", 
                            "maximum": 1.0
                        }, 
                        "fraction_radiant": {
                            "default": 0.0, 
                            "minimum": 0.0, 
                            "type": "number", 
                            "maximum": 1.0
                        }
                    }, 
                    "type": "object"
                }
            }, 
            "legacy_idd": {
                "numerics": {
                    "fields": [
                        "design_level", 
                        "watts_per_zone_floor_area", 
                        "watts_per_person", 
                        "fraction_latent", 
                        "fraction_radiant", 
                        "fraction_lost"
                    ]
                }, 
                "fields": [
                    "name", 
                    "zone_or_zonelist_name", 
                    "schedule_name", 
                    "design_level_calculation_method", 
                    "design_level", 
                    "watts_per_zone_floor_area", 
                    "watts_per_person", 
                    "fraction_latent", 
                    "fraction_radiant", 
                    "fraction_lost", 
                    "end_use_subcategory"
                ], 
                "alphas": {
                    "fields": [
                        "name", 
                        "zone_or_zonelist_name", 
                        "schedule_name", 
                        "design_level_calculation_method", 
                        "end_use_subcategory"
                    ]
                }
            }, 
            "type": "object", 
            "memo": "Sets internal gains for electric equipment in the zone. If you use a ZoneList in the Zone or ZoneList name field then this definition applies to all the zones in the ZoneList."
        }, 
```

**Comment by mbadams5** [comment]
@JasonGlazer Yes I can include some examples.

Yes everything that is in the JDD is being used. What seems like redundant data is almost entirely due to supporting IDF format and the legacy IDD format within EnergyPlus source code still. Everything in the `legacy_idd` key is used by the InputProcessor to map the new key/value pair fields to its legacy IDD position. This is critical to manipulate the new JSON format input data into the existing format that `GetObjectItem` and other InputProcessor functions return. This allows us to not change ANY `GetFooInput` functions since they still interact with positional based data for now. Moving forward as the `GetFooInput` functions are refactored to use the new JSON-based input, we eliminate the need for everything in the `legacy_idd` key in the JDD. 

Also this may be unclear from my initial description, but the JDD is automatically generated from the Energy+.idd file. That means developers will continue to manipulate, change, and add new features to the Energy+.idd file until such point that IDF/IDD is no longer supported. Until that point, the IDD is still the source of truth for EnergyPlus from which the JDD is generated. Along the same lines, EnergyPlus will still read and process IDF exactly the same as it has previously. This was done to ensure compatibility with existing tools and workflows while giving time to transition to the new format.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 21 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 16 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
Since I think the plan for now is to create a branch of IDD only changes and get those in directly, this branch won't *technically* have any IDD changes when it's ready to go later/soon.  So I'm removing the IDDChange label from it.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
@mbadams5 What's the status of this particular PR?  When the JSON is ready to drop, are we working off of the PR that has the output changes too?  In which case, is this one OK to just close?

**Comment by nrel-bot-3** [comment]
@mbadams5 @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
:+1: :+1: 

I'm scanning through the diffs, and so far everything looks great.  

#### Error File
- The error file diffs seem to be almost entirely just the field name casing change.
- There are a few where the number of unused objects changed.  If you look at the 5ZoneDetailedIceStorage, for example, the number of unused objects went from 1 to 6.  I built this branch and develop locally, added the unused diagnostic output, and found that both branches are behaving the same, just reporting different (better) now.  The file has 6 `Output:TableTimeBins` objects.  In each branch, the diagnostic for unused objects says it only reports the name of the first instance of each class.  However, in develop, it also says there's only 1 unused object.  But really there are 6, and that's what this branch shows. And when you add in the 'all warnings' diagnostic, the JSON branch shows the object name, which is helpful.
- I continued spot checking about half of the error file diffs and they all look great.

#### DXF File
- One file had DXF diffs: DaylightingDeviceShelf.  From what I can tell, the two DXFs are actually the same, just reordered:

image

I'll continue slogging through some diffs, but overall I have a very good impression of this.

**Comment by Myoldmopar** [comment]
Here's an oddity...I started looking through the 8 files with big ESO diffs.  I ran `5ZoneFanCoilDOASCool` locally for both branches, and started comparing the outputs.  Zone temperatures, control, everything looks essentially identical.  Heating coil results are good.  Cooling coil I saw diffs though.  Here is the first variable I saw, the total cooling rate:

image

A little odd that it would show diffs like that when the zone is maintaining control so well in both cases.  So I plotted the variable next to it, the **sensible** cooling rate on the same coil:

image

Yeah, so that explains why the zone conditions look very similar.  The coil is still behaving but something is causing the SHR to be different.  Three FanCoil example files all have ESO big diffs, and I bet whatever is causing this will clean up at least all three of those, maybe others.  Not sure if it's a curve input issue or rated SHR or something else.  I won't realistically have time to look at it again until tonight.

**Comment by mbadams5** [comment]
@Myoldmopar It could be a merge conflict resolution gone bad at some point, but that is still odd results.

**Comment by Myoldmopar** [comment]
@mbadams5 Some deep detective work revealed the culprit for at least a good chunk of the ESO diffs.  I'm pushing up the fix now.

Here's the sensible/total plots now:

image

They aren't exactly identical, but they are within like 6 W of each other throughout the day.

**Comment by Myoldmopar** [comment]
Actually I won't push the fix up because there is one merge conflict and I don't want to get involved.  Here's the change that needs to be applied so that it matches develop:

Line 3332 of your [branch](https://github.com/NREL/EnergyPlus/blob/bae473974a3e2e79599a6de86246c87eff4f8629/src/EnergyPlus/WaterCoils.cc#L3332):
```
-   TotWaterCoilLoad = WaterCoil( CoilNum ).InletAirMassFlowRate * ( EnthAirInlet - EnthAirOutlet );
+   TotWaterCoilLoad = AirMassFlow * ( EnthAirInlet - EnthAirOutlet );
```

File: src/EnergyPlus/HeatPumpWaterToWaterHEATING.cc [hunk 6]
```diff
@@ -116,10 +116,10 @@ namespace HeatPumpWaterToWaterHEATING {
 	Real64 Power( 0.0 ); // power consumption Watts Joules/sec
 	Real64 QLoad( 0.0 ); // heat rejection from Load Side coil Joules
 	Real64 QSource( 0.0 ); // cooling capacity Joules
-	Real64 SourceSideWaterOutletTemp( 0.0 ); // Source Side outlet temperature �C
-	Real64 SourceSideWaterInletTemp( 0.0 ); // Source Side outlet temperature �C
-	Real64 LoadSideWaterOutletTemp( 0.0 ); // Source Side outlet temperature �C
-	Real64 LoadSideWaterInletTemp( 0.0 ); // Source Side outlet temperature �C
+	Real64 SourceSideWaterOutletTemp( 0.0 ); // Source Side outlet temperature �C
```
Comment (mjwitte) [inline src/EnergyPlus/HeatPumpWaterToWaterHEATING.cc]: @mbadams5 Walking through the diffs looking for any other stray lines like the one @Myoldmopar found in the coil. There's quite a few comment lines like this where the degree symbol has been changed and shows ? in a diamond now on githhub. Are these ok? I know they don't impact results.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 7]
```diff
@@ -14854,7 +14782,7 @@ Label900: ;
 		UnitConvSize = 115;
 		UnitConv.allocate( UnitConvSize );
 		UnitConv( 1 ).siName = "%";
-		UnitConv( 2 ).siName = "�C";
+		UnitConv( 2 ).siName = "°C";
```
Comment (mjwitte) [inline src/EnergyPlus/OutputReportTabular.cc]: But here the degree symbol has gone from ? to a proper degree symbol.

File: src/EnergyPlus/WaterCoils.cc [hunk 8]
```diff
@@ -3385,8 +3329,8 @@ namespace WaterCoils {
 		AirInletCoilSurfTemp = PsyTsatFnHPb( EnthSatAirCoilSurfaceEntryTemp, OutBaroPress );
 
 		// Calculate outlet air temperature and humidity from enthalpies and surface conditions.
-		TotWaterCoilLoad = AirMassFlow * ( EnthAirInlet - EnthAirOutlet );
-		OutletWaterTemp = WaterTempIn + TotWaterCoilLoad / max( WaterMassFlowRate, SmallNo ) / Cp;
+		TotWaterCoilLoad = WaterCoil( CoilNum ).InletAirMassFlowRate * ( EnthAirInlet - EnthAirOutlet );
+		OutletWaterTemp = WaterTempIn + TotWaterCoilLoad / max( WaterCoil( CoilNum ).InletWaterMassFlowRate, SmallNo ) / Cp;
```
Comment (mjwitte) [inline src/EnergyPlus/WaterCoils.cc]: @Myoldmopar I thought you said you pushed the fix for this. These two lines are still making a change that they shouldn't be. After I finish this pass I'll refresh and make sure.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 9]
```diff
@@ -5643,8 +5589,8 @@ namespace WaterThermalTanks {
 				HPWaterHeater( HPNum ).NumofSpeed = VarSpeedCoil( VSCoilID ).NumOfSpeeds;
 
 			}
-			else if ( SameString( HPWaterHeater( HPNum ).DXCoilType, "Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed" ) &&
-					  ( 0 == HPWaterHeater( HPNum ).NumofSpeed ) ) {
+			else if ( UtilityRoutines::SameString( HPWaterHeater( HPNum ).DXCoilType, "Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed" ) &&
+					  ( HPWaterHeater( HPNum ).NumofSpeed == 0 ) ) {
```
Comment (mjwitte) [inline src/EnergyPlus/WaterThermalTanks.cc:5593]: Well that was cute. `0 == . . . ` surprised that compiled.

**Review by mjwitte — COMMENTED** [review]
An impressive body of work. Reads like a good novel. Will refesh this and double-check the watercoil change after I submit this review.

**Comment by Myoldmopar** [comment]
@mjwitte No, I hadn't pushed, and @mbadams5 found the second line as well.  I ran his latest locally and verified that those two lines cleaned up the diffs **completely**.

**Comment by mjwitte** [comment]
@Myoldmopar OK - blew past that comment earlier.  So, the integration tests are all failing now, but you're probably seeing that.  I'll move on to other stuff.

**Comment by Myoldmopar** [comment]
I checked 1ZoneEvapCooler, which is now failing.  There is an array bounds error when the code is trying to assign a value to cAlphaArgs(1) in SizingManager.cc:1204, in GetSizingParams.  Presumably this is being called in every example file, which is why they fail.  But I've only tested the one file.

**Comment by Myoldmopar** [comment]
So I'm trying to track down the exact commit that caused this to go awry:

image

In that section of the commit log, b1fdd3a was showing good results.  Diffs, but successful runs.  By bae4739, the test files were failing.  I used git bisect to help narrow down between there, but things got hairy because an intermediate commit failed to build, so I wasn't sure whether to tell bisect that it was a good or bad commit.  I chose bad on the failing build commit and based on that it determined that 53d87e93 was the bad commit that introduced the error.  Since I had to guess on the failed build, it could be wrong.  And I don't see how that commit could've broken it, but maybe...

That's all I've got for now @mbadams5.

**Comment by Myoldmopar** [comment]
!!! OK, so reverting 53d87e9 out of the current branch state allows the simulation to run.  I don't know why, I'm sure @mbadams5 does.  If I need to, soon I'll push that reversion up to here so CI can verify what I'm seeing locally.

**Comment by Myoldmopar** [comment]
Verification about reverting success: https://gist.github.com/Myoldmopar/aaace5182721a1126654f620c53f55a7

**Comment by Myoldmopar** [comment]
Yeah! 1 Test has big ESO diffs!  I'm looking at those diffs now.

**Comment by Myoldmopar** [comment]
The LshapedZone example file that is showing ESO diffs is shown here compared to develop.

image

It isn't a big diff, but it is big enough to throw warnings.

**Comment by mjwitte** [comment]
@Myoldmopar Are the Std 62 changes present in both? I merged that into develop in the middle of all this.

**Comment by Myoldmopar** [comment]
I was just discussing that with @mbadams5.  That could definitely be a part of it, but I'm not sure I had pulled the 62.1 into my local develop yet.  He's going to be pushing up potentially "final" changes after he merges develop and resolves conflicts.  I'll pull immediately and compare to the latest develop at that time.

**Comment by mjwitte** [comment]
@mbadams5 There may be one more merge going into develop,shortly #6454 @Myoldmopar will decide.

**Comment by Myoldmopar** [comment]
Good point @mjwitte.  As soon as @nrel-bot-3 gets done with its build of 6454 and it looks good, I'll merge it in.  That will shut the door to everything except JSON.

**Comment by mjwitte** [comment]
@mbadams5 Is \obsolete supported here?  Some files have one less warning [now](http://energyplus.s3-website-us-east-1.amazonaws.com/regressions/2018-02/e592307cd22e62e47b171dcea78204ddb1143faf-b5d3be20dfab22f20c80fd271e9c2f128e85acc9/1ZoneParameterAspect/x86_64-MacOS-10.9-clang/eplusout.err.diff.html)
```
-   ** Warning ** IP: IDF line~527 Obsolete object=AirTerminal:SingleDuct:Uncontrolled, encountered.  Should be replaced with new object=AirTerminal:SingleDuct:ConstantVolume:NoReheat
```

**Comment by Myoldmopar** [comment]
I verified locally that single remaining ESO diff is now gone.

**Comment by Myoldmopar** [comment]
Good to go!  Merging!!

**Pull request merged**